main/uart_nic.c

```diff
@@ -386,7 +460,9 @@ static void check_online_status() {
     const uint32_t elapsed = now >= last ? now - last : now;
 
     if (elapsed > INACTIVE_PACKET_SECONDS) {
-        esp_wifi_disconnect();
+        probe_in_progress = true;
```

Comment:
It's not needed here, because the `check_online_status` is ever called only from one thread. But as a general practice, I tend to do some kind of atomic exchange/swap instead and rely on the guarantee from this function not being thread safe.